scripts/pdf/render.py

```diff
@@ -15,106 +15,66 @@
-
-    # Creating title page
-    with open("title.html", "w") as f:
-        f.write(header + title_content)
+        + '<p style="page-break-before: always" ></p>'
 
-    group.append(HTML("title.html").render())
+    # Writing names of all directories inside 'pages' to a list
+    for os_dir in os.listdir(loc):
+        oslist.append(os_dir)
+    oslist.sort()
 
     for operating_sys in oslist:
-
         # Required string to create directory title pages
-        dir_title = "<h2 class=title-dir>" + \
-            operating_sys.capitalize() + "</h2></body></html>"
-
-        # Creating directory title page for current directory
-        with open("dir_title.html", "w") as os_html:
-            os_html.write(header + dir_title)
-
-        group.append(HTML("dir_title.html").render())
+        html += "<h2 class=title-dir>" + operating_sys.capitalize() + "</h2>" \
+            + '<p style="page-break-before: always" ></p>'
 
         # Creating a list of all md files in the current directory
         for temp in glob.glob(os.path.join(loc, operating_sys, "*.md")):
             allmd.append(temp)
-
-        # Sorting all filenames in the directory, to maintain the order of the PDF
         allmd.sort()
 
-        # Conversion of Markdown to HTML
+        # Conversion of Markdown to HTML string
         for page_number, md in enumerate(allmd, start=1):
-
-                with open(md, "r") as inp:
-                    text = inp.readlines()
-
-                with open("htmlout.html", "w") as out:
-                    out.write(header)
-
-                    for line in text:
-                        if re.match(r'^>', line):
-                            line = line[:0] + '####' + line[1:]
-                        html = markdown.markdown(line)
-                        out.write(html)
-                    out.write(footer)
-
-                group.append(HTML("htmlout.html").render())
-                print("Rendered page {} of the directory {}".format(
-                    str(page_number), operating_sys))
-
+            with open(md, "r") as inp:
+                text = inp.readlines()
+                for line in text:
+                    if re.match(r'^>', line):
+                        line = line[:0] + '####' + line[1:]
+                    html += markdown.markdown(line)
+            html += '<p style="page-break-before: always" ></p>'
+            print("Rendered page {} of the directory {}".format(
+                str(page_number), operating_sys))
         allmd.clear()
-
-    # Merging all the documents into a single PDF
-    for doc in group:
-        for p in doc.pages:
-            ap.append(p)
-
-    # Writing the PDF to disk, preserving metadata of first `tldr` page
-    group[2].copy(ap).write_pdf('tldr-pages.pdf')
+    
+    html += "</body></html>"
+    
+    # Writing the PDF to disk
+    print("\nConverting all pages to PDF...")
+    HTML(string=html).write_pdf("tldr-pages.pdf", stylesheets=csslist)
```

Comment:
From the [api reference](https://doc.courtbouillon.org/weasyprint/stable/api_reference.html#weasyprint.HTML.write_pdf) for this function, could `csslist` be a list of filename strings instead of instantiated `CSS` classes?

Reply:
Yes, it could. I didn't notice that `stylesheets` accepts string list.